<?xml version="1.0"?>
<launch>
  <arg name="gdb_debug" default="false"/>
  <arg name="profile" default="false"/>

  <arg name="data_set_path" default="/home/ken/datasets/euroc/MH_03/mav0"/>
  <arg name="stand_still_start" default="1403637142538319104."/>
  <arg name="stand_still_end"   default="1403637149088318946."/>

  <group ns="msckf">
    <param name="data_set_path" value="$(arg data_set_path)"/>
    <param name="stand_still_start" value="$(arg stand_still_start)"/>
    <param name="stand_still_end" value="$(arg stand_still_end)"/>

    <param name="feature_cov" value="2"/>

    <param name="imu_vars/w_var" value="1e-4"/>
    <param name="imu_vars/dbg_var" value="3.6733e-5"/>
    <param name="imu_vars/a_var" value="1e-2"/>
    <param name="imu_vars/dba_var" value="7e-2"/>

    <param name="imu_covars/p_init" value="1e-12"/>
    <param name="imu_covars/q_init" value="1e-5"/>
    <param name="imu_covars/v_init" value="1e-2"/>
    <param name="imu_covars/ba_init" value="1e-2"/>
    <param name="imu_covars/bg_init" value="1e-2"/>

    <param name="translation_threshold" value="0.01"/><!--Max translation from marginalizing state-->

    <param name="keyframe_transl_dist" value="0.05"/>
    <param name="keyframe_rot_dist" value="0.05"/>

    <param name="min_track_length" value="3"/>
    <param name="max_track_length" value="50"/>
    <param name="max_cam_states" value="30"/>

    <param name="ransac_threshold" value="0.00000002"/>

    <param name="n_grid_rows" value="10"/>
    <param name="n_grid_cols" value="10"/>

    <group unless="$(arg profile)">
      <node pkg="msckf_mono" name="asl_msckf" type="asl_msckf_no_ground_truth" unless="$(arg gdb_debug)" output="screen"/>
      <node pkg="msckf_mono" name="asl_msckf" type="asl_msckf_no_ground_truth" launch-prefix="xterm -e gdb --args" if="$(arg gdb_debug)"/>
    </group>                                                            
    <group if="$(arg profile)">                                         
      <node pkg="msckf_mono" name="asl_msckf" type="asl_msckf_no_ground_truth" launch-prefix="perf record -g --call-graph dwarf --output=perf.out.node_name.data --" if="$(arg profile)"/>
    </group>

    <node pkg="rviz" name="msckf_rviz" type="rviz" args="-d $(find msckf_mono)/launch/asl_msckf.rviz"/>

  </group>
</launch>
